src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1108,6 +1125,282 @@ bool os::dll_address_to_library_name(address addr, char* buf,
   return true;
 }
 
+// get the library search path burned in to the executable file during linking
+// If the libpath cannot be retrieved return an empty path
+static const char* rtv_linkedin_libpath() {
+  static char buffer[4096];
+  static const char* libpath = 0;
```

Comment:
If your intent is to return an empty buffer if there is no contained libpath, I would just:

```
static const char* libpath = "";
```
then you can always just return libpath.

Reply:
But looking at the using code, returning NULL in case there is no contained libpath would be actually easier, see below.